pythonFiles/include/blender_vscode/load_addons.py

```diff
@@ -1,41 +1,64 @@
 import os
-import bpy
 import sys
 import traceback
 from pathlib import Path
-from . communication import send_dict_as_json
-from . environment import user_addon_directory, addon_directories
 
-def setup_addon_links(addons_to_load):
-    if not os.path.exists(user_addon_directory):
-        os.makedirs(user_addon_directory)
+import bpy
+
+from . import AddonInfo
+from .communication import send_dict_as_json
+from .environment import addon_directories
+from .utils import is_addon_legacy
+
 
-    if not str(user_addon_directory) in sys.path:
-        sys.path.append(str(user_addon_directory))
+def setup_addon_links(addons_to_load: list[AddonInfo]):
 
     path_mappings = []
 
-    for source_path, module_name in addons_to_load:
-        if is_in_any_addon_directory(source_path):
-            load_path = source_path
+    for addon_info in addons_to_load:
+        user_addon_directory = get_user_addon_directory(Path(addon_info.load_dir))
+        print(f"USER ADDON: {user_addon_directory}")
+
+        if not os.path.exists(user_addon_directory):
+            os.makedirs(user_addon_directory)
+
+        if not str(user_addon_directory) in sys.path:
+            sys.path.append(str(user_addon_directory))
```

Comment:
I guess this shouldn't be done for extensions anymore? (haven't looked into it much yet)

Reply:
Everything seems to work properly when this is disabled for extensions, so it's probably not necessary?
I wasn't sure exactly what it was for, so I just left it in the original request.